src/logics/UFLRATheory.cc

```diff
@@ -1,21 +1,29 @@
 #include "UFLRATheory.h"
+#include "TreeOps.h"
 #include "ArithmeticEqualityRewriter.h"
 
 bool UFLRATheory::simplify(const vec<PFRef>& formulas, PartitionManager &pmanager, int curr)
 {
     auto & currentFrame = pfstore[formulas[curr]];
     // Take care of UF simplifications as well
+    ArithmeticEqualityRewriter rewriter(lralogic);
```

Comment:
This rewriter is not used, the branches still create their own rewriters. Also, `notOkToParition` is assigned only in the if branch, but not in the else branch.

Reply:
Right, I fixed this and tried to make also the LATheory simplification a little more economic in lines of code.